File: stress/t/arr_bubble.f
      PROGRAM ARRBUBBLE
      INTEGER*8 I,J,N,A,B,T,SEED,OK,C1
      INTEGER*8 AR(0:15)
      N=16
      SEED=12345
      C1=1103515245
      I=0
   10 IF (I.GE.N) GOTO 20
      SEED=SEED*C1+12345
      T=MOD(SEED,1000_8)
      IF (T.LT.0) T=T+1000
      AR(I)=T
      I=I+1
      GOTO 10
   20 CONTINUE
      I=0
   30 IF (I.GE.N-1) GOTO 60
      J=0
   40 IF (J.GE.N-1-I) GOTO 50
      A=AR(J)
      B=AR(J+1)
      IF (A.GT.B) THEN
        AR(J)=B
        AR(J+1)=A
      ENDIF
      J=J+1
      GOTO 40
   50 CONTINUE
      I=I+1
      GOTO 30
   60 CONTINUE
      OK=1
      I=0
   70 IF (I.GE.N-1) GOTO 80
      A=AR(I)
      B=AR(I+1)
      IF (A.GT.B) OK=0
      I=I+1
      GOTO 70
   80 CONTINUE
      WRITE(*,'(A,I0)') 'sorted=',OK
      T=AR(0)
      WRITE(*,'(A,I0)') 'min=',T
      T=AR(15)
      WRITE(*,'(A,I0)') 'max=',T
      END

File: stress/t/arr_bubble.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,j,n,a,b,t,seed,ok;
  int64_t A[16];
  n=16;
  seed=12345;
  for(i=0;i<n;i++){
    seed=(int64_t)((uint64_t)seed*(uint64_t)1103515245 + (uint64_t)12345);
    t=seed%1000;
    if(t<0){ t=t+1000; }
    A[i]=t;
  }
  for(i=0;i<n-1;i++){
    for(j=0;j<n-1-i;j++){
      a=A[j]; b=A[j+1];
      if(a>b){ A[j]=b; A[j+1]=a; }
    }
  }
  ok=1;
  for(i=0;i<n-1;i++){ a=A[i]; b=A[i+1]; if(a>b){ ok=0; } }
  printf("sorted="); printf("%ld",(long)ok); printf("\n");
  t=A[0]; printf("min="); printf("%ld",(long)t); printf("\n");
  t=A[15]; printf("max="); printf("%ld",(long)t); printf("\n");
  return 0;
}
